modules/consent/www/getconsent.php

```diff
@@ -139,16 +183,101 @@
     break;
 }
 
-if (array_key_exists('consent:store', $state)) {
-    $t->data['usestorage'] = true;
-} else {
-    $t->data['usestorage'] = false;
-}
+$t->data['usestorage'] = array_key_exists('consent:store', $state);
 
 if (array_key_exists('consent:hiddenAttributes', $state)) {
     $t->data['hiddenAttributes'] = $state['consent:hiddenAttributes'];
 } else {
     $t->data['hiddenAttributes'] = array();
 }
 
+$t->data['attributes_html'] = present_attributes($t);
+
 $t->show();
+
+/**
+ * Recursive attribute array listing function
+ *
+ * @param SimpleSAML_XHTML_Template $t          Template object
+ *
+ * @return string HTML representation of the attributes
+ */
+function present_attributes($t)
+{
+    $translator = $t->getTranslator();
+
+    $alternate = array('odd', 'even');
+    $i = 0;
+    $summary = 'summary="' . $t->t('{consent:consent:table_summary}') . '"';
+
+    $str = '<table id="table_with_attributes" class="attributes" '. $summary .'>';
+    $str .= "\n" . '<caption>' . $t->t('{consent:consent:table_caption}') . '</caption>';
+
+    $attributes = $t->data['attributes'];
+    foreach ($attributes as $name => $value) {
+        $nameraw = $name;
+        $name = $translator->getAttributeTranslation($nameraw);
+
+        if (preg_match('/^child_/', $nameraw)) {
+            // insert child table
+            $parentName = preg_replace('/^child_/', '', $nameraw);
+            foreach ($value as $child) {
+                $str .= "\n" . '<tr class="odd"><td class="td_odd">' .
+                    present_attributes($t, $child, $parentName) . '</td></tr>';
```

Comment:
I removed the 2nd and 3rd argument because (i thought) they were never used... Obviously I've missed something 💩 and my tests missed the nested attribute-table case...
I'll revert this and see if we can use the function from `templates/includes/attributes.php`